--- configure.ac.old	2016-09-17 12:04:50.021130544 -0700
+++ configure.ac	2016-09-17 13:01:51.575663415 -0700
@@ -32,11 +32,12 @@
 AC_PATH_PROG([CURL], [curl], [no])
 AC_PATH_PROG([WGET], [wget], [no])
 AC_PATH_PROG([FTP], [ftp], [no])
-AS_IF([test x"$CURL" != xno], [FETCHER="$CURL -o - --ftp-pasv"],
-	[test x"$WGET" != xno], [FETCHER="$WGET -O -"],
-	[test x"$FTP" != xno], [FETCHER="$FTP -o - -p"],
+AS_IF([test x"$CURL" != xno], [FETCHER="$CURL --ftp-pasv"; FETCHER_OUTPUT="-o"],
+	[test x"$WGET" != xno], [FETCHER="$WGET"; FETCHER_OUTPUT="-O"],
+	[test x"$FTP" != xno], [FETCHER="$FTP -p"; FETCHER_OUTPUT="-o"],
 	[AC_MSG_ERROR([no file transfer utility found])])
 AC_SUBST(FETCHER)
+AC_SUBST(FETCHER_OUTPUT)
 
 AC_ARG_ENABLE(linux,
         [AS_HELP_STRING([--enable-linux],
@@ -60,7 +61,6 @@
 	[AS_HELP_STRING([--with-arch=IMAFD],
 		[Sets the base RISC-V ISA, defaults to IMAFD])],
 	AC_SUBST(WITH_ARCH, --with-arch=$withval),
-	AC_SUBST(WITH_ARCH,)
 	)
 
 AC_ARG_ENABLE(atomic,
@@ -95,7 +95,7 @@
 	)
 
 AS_IF([test "x$enable_multilib" != xno],
-        [AC_SUBST(multilib_flags,)],
+        ,
 	[AC_SUBST(multilib_flags,--disable-multilib)])
 
 AC_CONFIG_FILES([Makefile])
